src/io/routes.js

```diff
@@ -0,0 +1,253 @@
+import { formatDir, format404 } from './html-formatter';
+
+const ioInRegex = /\/io\/in(\/.*)/;
+const ioResetRegex = /\/io\/reset/;
+const ioRemoveRegex = /\/io\/remove\/(.+)/;
+const ioToRegex = /\/io\/to\/dataurl(\/.*)/;
+const ioOutRegex = /\/io\/out(\/.*)/;
+const ioImportRegex = /\/io\/import/;
+const ioFromTextRegex = /\/io\/from\/text(\/.*)/;
+const ioFromDataURIRegex = /\/io\/from\/dataurl(\/.*)/;
+
+export default (workbox, ioServer) => {
+    workbox.routing.registerRoute(
+        ioInRegex,
+        async ({ url }) => {
+            const path = decodeURIComponent(url.pathname.match(ioInRegex)[1]);
+            let body, type, status;
+            try {
+                const res = await ioServer.createPath(path);
+                const result = await ioServer.serve(path);
+
+                body = result.body;
+                type = result.type;
+                status = 200;
+            } catch (err) {
+                body = err;
+                type = 'text/html';
+                // TODO: should probably do a better job here on mapping to err
+                status = 404;
+            }
+
+            const init = {
+                status,
+                statusText: 'OK',
+                headers: { 'Content-Type': type },
+            };
+
+            return new Response(body, init);
+        },
+        'GET'
+    );
+
+    workbox.routing.registerRoute(
+        ioRemoveRegex,
+        async ({ url }) => {
+            const path = '/' + decodeURIComponent(url.pathname.match(ioRemoveRegex)[1]);
+            let body, type, status;
+            try {
+                const result = await ioServer.deletePathRecursively(path);
+                return Response.redirect(`${url.origin}/io/in/`);
+            } catch (err) {
+                body = err;
+                type = 'text/html';
+                // TODO: should probably do a better job here on mapping to err
+                status = 404;
+            }
+
+            const init = {
+                status,
+                statusText: 'OK',
+                headers: { 'Content-Type': type },
+            };
```

Comment:
Your functions in here duplicate a ton of code.  You should refactor common things into a set of functions you call.